notebooks/05_operationalize/aks_locust_load_test.ipynb

```diff
@@ -0,0 +1,392 @@
+{
```

Comment:
<p>Not sure if we are being to low here. In the [first reference architecture](<a href="https://docs.microsoft.com/en-us/azure/architecture/reference-architectures/ai/real-time-recommendation" target="_blank">https://docs.microsoft.com/en-us/azure/architecture/reference-architectures/ai/real-time-recommendation</a>) we had 200 concurrent users and 180 requests per second. Here we have 10 concurrent users and 91 req/s. </p><p>I think it would be great to have some numbers in the order of magnitude of what our customers would like. Maybe we can ask the rest of the team about this</p>

 _Reply via <a href='https://app.reviewnb.com/Microsoft/Recommenders/pull/758/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Reply:
k, i'll boost this up to 200 concurrent users

Reply:
I'm looking at the results, we are getting some errors:
```
Error report
 # occurrences      Error                                                                                               
--------------------------------------------------------------------------------------------------------------------------------------------
 11                 POST /api/v1/service/lightgbm-criteo/score: "HTTPError('503 Server Error: Service Unavailable for url: http://13.92.193.6:80/api/v1/service/lightgbm-criteo/score',)"
```
Do you think it is because the AKS cluster is not big enough?

Reply:
most likely yes, I had to change my deployment to disable auto-scaling, because I was hitting more errors than this. It's a very small percentage though, and I think is actually good to see what the errors look like in this example. But we could always adjust the deployment notebook to boost the replicas to avoid this.

Reply:
I'm ok with either. In fact this is something that Tao might want to review. I'll add this discussion to teams

Reply:
Scott what AKS size were you using for the testing? Btw autoscaling doesn't play well with (short) load tests, autoscaling on AKS takes a min 5 minutes to trigger autoscale (+ the time it will take to create a new node).

Reply:
this was using a cluster of 3 nodes, with max replicas of 15, details on setup are in this operationalization notebook: https://github.com/Microsoft/Recommenders/blob/master/notebooks/05_operationalize/lightgbm_criteo_o16n.ipynb